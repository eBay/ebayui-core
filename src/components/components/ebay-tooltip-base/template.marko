<span
    w-on-expander-expand="handleExpand"
    w-on-expander-collapse="handleCollapse"
    w-body
    overlay-style=data.overlayStyle
    w-bind/>
