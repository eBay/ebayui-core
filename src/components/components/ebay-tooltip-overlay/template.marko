<script marko-init>
    var constants = require("./constants");
    var typeRoles = constants.typeRoles;
    var pointerStyles = constants.pointerStyles;
</script>

<var hasUserStyles=(
    data.styleTop ||
    data.styleLeft ||
    data.styleRight ||
    data.styleBottom
)/>
<var overlayStyles=(hasUserStyles ? {
    top: data.styleTop,
    left: data.styleLeft,
    right: data.styleRight,
    bottom: data.styleBottom
} : pointerStyles[data.pointer || "bottom"])/>

<div
    class="${data.type}__overlay"
    role=typeRoles[data.type]
    aria-labelledby=(data.type === "tourtip" && widget.elId("tourtip-label"))
    id=data.tooltipId
    style=overlayStyles
    w-bind>
    <span class="${data.type}__pointer ${data.type}__pointer--${data.pointer}"/>
    <div class="${data.type}__mask">
        <div class="${data.type}__cell">
            <div class="${data.type}__content">
                <if(data.heading)>
                    <h2
                        class=["${data.type}__heading", data.heading.class]
                        style=data.heading.style
                        w-id="tourtip-label"
                        w-body=data.heading.renderBody/>
                </if>

                <if(data.content)>
                    <div
                        body-only-if(!data.content.class && !data.content.style)
                        class=data.content.class
                        style=data.content.style
                        w-body=data.content.renderBody/>
                </if>
            </div>

            <if(data.type !== "tooltip")>
                <button
                    aria-label=data.a11yCloseText
                    class="${data.type}__close"
                    type="button"
                    w-on-click="handleCloseButton">
                    <span aria-hidden="true" role="img"/>
                </button>
            </if>
        </div>
    </div>
</div>
