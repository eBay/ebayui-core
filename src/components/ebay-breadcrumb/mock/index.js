const getItem = (text, href) => {
    const item = {
        href: 'https://www.ebay.com/',
        navSrc: '{"actionKind":"NAVSRC","operationId":"2489527"}',
        _sp: 'p2489527.m4340.l9751.c1'
    };
    const clonedItem = JSON.parse(JSON.stringify(item));
    clonedItem.renderBody = (stream) => stream.write(text);
    clonedItem.href = href;
    return clonedItem;
};
const basicItems = [getItem('eBay', 'https://www.ebay.com'),
    getItem('Auto Parts and Vehicles', 'https://www.ebay.com/b/Auto-Parts-and-Vehicles/6000/bn_1865334'),
    getItem('Motors Parts & Accessories', null)];

const firstItemMissingHref = [getItem('eBay', null),
    getItem('Auto Parts and Vehicles', 'https://www.ebay.com/b/Auto-Parts-and-Vehicles/6000/bn_1865334'),
    getItem('Motors Parts & Accessories', null)];

module.exports = {
    basicItems: {
        headingText: 'Page navigation',
        items: basicItems
    },
    hijax: {
        hijax: true,
        items: basicItems
    },
    firstItemMissingHref: {
        headingText: 'Page navigation',
        items: firstItemMissingHref
    },
    itemsWithHeadingLevel: {
        headingText: 'Page navigation',
        headingLevel: 'h3',
        items: basicItems
    }
};
