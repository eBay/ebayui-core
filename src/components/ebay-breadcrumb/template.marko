<script marko-init>
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<nav w-bind
    aria-labelledby=widget.elId("breadcrumb-heading")
    class=["breadcrumb", data.class]
    style=data.style
    role="navigation"
    ${processHtmlAttributes(data)}>

    <${data.a11yHeadingTag || "h2"} id=widget.elId("breadcrumb-heading") class="clipped">
        ${data.a11yHeadingText}
    </>
    <ul>
        <li for(item in data.items | status-var="status")>
            <var current=(!item.href && status.isLast())/>
            <${item.href ? "a" : "button"}
                href=item.href
                aria-current=(current && "location")
                class=item.class
                style=item.style
                w-body=item.renderBody
                w-onclick=(!current && "handleClick")
                ${processHtmlAttributes(item)}/>
        </li>
    </ul>
</nav>
