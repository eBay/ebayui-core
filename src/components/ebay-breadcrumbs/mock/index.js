import { createRenderBody, getNItems } from '../../../common/test-utils/shared';

export const Links = {
    a11yHeadingText: 'Page navigation',
    items: getNItems(3, (i) => ({
        href: i === 2 ? undefined : '#',
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const Buttons = {
    items: getNItems(3, (i) => ({
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const linkLastWithoutHREF = {
    a11yHeadingText: 'Page navigation',
    items: getNItems(3, (i) => ({
        href: i === 2 ? undefined : '#',
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const linkHeadingTag = Object.assign({}, Links, {
    a11yHeadingTag: 'h3',
});
