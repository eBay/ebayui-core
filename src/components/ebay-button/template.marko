<script marko-init>
    var getWidgetId = require("../../common/get-marko-3-widget-id");
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<var size=(data.size || "medium")/>
<var priority=(data.priority || "secondary")/>
<var variant=(!data.variant && data.href ? "fake" : data.variant)/>
<var isIconVariant=(variant === 'icon')/>
<var isExpandVariant=(variant === 'expand')/>
<var isCtaVariant=(variant === 'cta')/>
<var isBadged=Boolean(data.badgeNumber && isIconVariant)/>
<var noText=(isIconVariant || isBadged || (isExpandVariant && data.noText))/>
<var baseClass=(variant ? "${variant}-btn" : "btn")/>
<var sizeClass="${baseClass}--${size}"/>
<var htmlAttributes=processHtmlAttributes(data)/>
<var tag=(data.href ? "a" : "button")/>

<${tag}
    ${htmlAttributes}
    w-bind
    w-onclick="handleClick"
    w-onkeydown="handleKeydown"
    id=(data.id || getWidgetId(out))
    class=[
        data.class,
        baseClass,
        noText && "${baseClass}--no-text",
        isBadged && "${baseClass}--badged",
        data.fluid && "${baseClass}--fluid",
        data.truncate && "${sizeClass}-truncated",
        data.fixedHeight && "${sizeClass}-fixed-height",
        !data.fixedHeight && !data.truncate && sizeClass,
        (priority === "secondary" || priority === "primary") && "${baseClass}--${priority}"
    ]
    style=data.style
    href=data.href
    type=(tag === "button" && data.type || "button")
    disabled=data.disabled
    aria-disabled=(data.partiallyDisabled && "true")>

    <if(isCtaVariant)>
        <span class="cta-btn__cell">
            <span w-body/>
            <ebay-icon type="inline" name="arrow-right-bold" class="cta-btn__icon" no-skin-classes/>
        </span>
    </if>
    <else>
        <var hasAriaLabel=Boolean(htmlAttributes.ariaLabel)/>
        <span body-only-if(!hasAriaLabel) w-body aria-hidden="true"/>
        <if(isBadged)>
            <ebay-badge
                number=data.badgeNumber
                type="icon"
                aria-label=(hasAriaLabel && data.badgeAriaLabel)
                aria-hidden=(hasAriaLabel && "true")/>
        </if>
    </else>
</>
