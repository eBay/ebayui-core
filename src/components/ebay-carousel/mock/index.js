const renderBody = (stream) => stream.write('text');
const itemWidth = 200;
const item = { renderBody, '*': { style: `width:${itemWidth}px` } };
const threeItems = [item, item, item];
const sixItems = threeItems.concat(threeItems);
const twelveItems = sixItems.concat(sixItems);

// mocks for visual debugging
const debugItem = { renderBody, '*': { style: `height:200px;width:400px;background:gray` } };
const debugItems = [debugItem, debugItem, debugItem, debugItem, debugItem, debugItem];

module.exports = { renderBody, itemWidth, item, threeItems, sixItems, twelveItems, debugItems };
