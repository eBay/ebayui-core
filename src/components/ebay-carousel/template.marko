<div class=data.classes style=data.style w-bind ${data.htmlAttributes}>
    <var config=data.config/>
    <var statusId=("carousel-status-" + widget.id)/>
    <div w-id="container" class=["carousel__container", data.bothControlsDisabled && "carousel__container--controls-disabled"]>
        <${data.a11yStatusTag}
            if(data.totalSlides >= 1)
            id=statusId
            class="clipped"
            aria-live=(data.autoplayInterval && !data.paused ? "off" : "polite")>
            <span>${data.a11yStatusText}</span>
        </>
        <button
            class="carousel__control carousel__control--prev"
            w-onclick=(!data.prevControlDisabled && "handleMove")
            data-direction=-1
            aria-describedby=statusId
            aria-label=data.a11yPreviousText
            aria-disabled=(data.prevControlDisabled && 'true')>
            <ebay-icon type="inline" name="chevron-left"/>
        </button>

        <ul w-id="list"
            class="carousel__list"
            style=(!config.nativeScrolling && data.offset && {
                transform:  "translate3d(" + (data.offset * -1) + "px,0,0)",
                transition: data.disableTransition ? 'none' : undefined
            })>
            <for(item in data.items)>
                <li ${item.htmlAttributes}
                    class=item.class
                    style=item.style
                    w-body=item.renderBody
                    aria-hidden=(!item.fullyVisible && 'true')
                    w-preserve-body-if(!!config.preserveItems)/>
            </for>
        </ul>

        <button
            w-id="next"
            class="carousel__control carousel__control--next"
            w-onclick=(!data.nextControlDisabled && "handleMove")
            data-direction=1
            aria-describedby=statusId
            aria-label=data.a11yNextText
            aria-disabled=(data.nextControlDisabled && 'true')>
            <ebay-icon type="inline" name="chevron-right"/>
        </button>

        <if(data.autoplayInterval && !data.bothControlsDisabled)>
            <if(data.paused)>
                <button
                    class="carousel__play"
                    aria-label=data.a11yPlayText
                    w-onclick="togglePlay">
                    <ebay-icon type="inline" name="play"/>
                </button>
            </if>
            <else>
                <button
                    class=`carousel__pause`
                    aria-label=data.a11yPauseText
                    w-onclick="togglePlay">
                    <ebay-icon type="inline" name="pause"/>
                </button>
            </else>
        </if>
    </div>

    <ul class="carousel__dots" if(!data.noDots && data.totalSlides > 1)>
        <li for(i from 0 to data.totalSlides - 1)>
            <var isActive=(i === data.slide)/>
            <button
                class={"carousel__dot--active": isActive}
                w-onclick=(!isActive && "handleDotClick")
                data-slide=i
                aria-describedby=statusId
                aria-disabled=(isActive && 'true')
                aria-label=(isActive ?
                    data.a11yCurrentText.replace('{currentSlide}', i + 1) :
                    data.a11yOtherText.replace('{slide}', i + 1))/>
        </li>
    </ul>
</div>
