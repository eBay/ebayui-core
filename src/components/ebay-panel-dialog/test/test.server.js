import { describe, it, expect } from "vitest";

import { render } from "@marko/testing-library";
import { testPassThroughAttributes } from "../../../common/test-utils/server";
import template from "../index.marko";
import * as mock from "./mock";

describe("dialog", () => {
    it("renders basic version", async () => {
        const input = mock.Dialog;
        const { getByRole, getByLabelText, getByText } = await render(
            template,
            input,
        );
        const dialog = getByRole("dialog", { hidden: true });

        expect(dialog).toMatchSnapshot();
        expect(getByLabelText(input.a11yCloseText)).toMatchSnapshot();
        expect(getByText(input.renderBody.text)).toMatchSnapshot();
    });

    it("renders with header and footer", async () => {
        const input = mock.headerFooterDialog;
        const { getByRole, getByLabelText, getByText } = await render(
            template,
            input,
        );
        const dialog = getByRole("dialog", { hidden: true });

        expect(dialog).toMatchSnapshot();
        expect(getByLabelText(input.a11yCloseText)).toMatchSnapshot();
        expect(getByText(input.renderBody.text)).toMatchSnapshot();
        expect(
            getByText(input.header.renderBody.text).parentElement,
        ).toMatchSnapshot();
        expect(getByText(input.footer.renderBody.text)).toMatchSnapshot();
    });

    it("renders in open state", async () => {
        const input = mock.dialogOpen;
        const { getByRole } = await render(template, input);
        expect(getByRole("dialog")).toMatchSnapshot();
    });

    it(`renders with end type`, async () => {
        const { getByRole } = await render(template, {
            position: "end",
            open: true,
        });
        const $dialog = getByRole("dialog");
        const $window = $dialog.children[0];

        expect($dialog).toMatchSnapshot();
        expect($window).toMatchSnapshot();
    });

    testPassThroughAttributes(template);
});
