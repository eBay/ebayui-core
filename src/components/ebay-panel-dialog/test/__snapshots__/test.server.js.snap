// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dialog > passes through additional html attributes 1`] = `
<div
  aria-modal="true"
  class="panel-dialog class1 panel-dialog--mask-fade-slow"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  hidden=""
  role="dialog"
  style="color:red"
  type="number"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#14-->
      <button
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > passes through additional html attributes 2`] = `
<div
  aria-modal="true"
  class="panel-dialog class1 panel-dialog--mask-fade-slow"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  hidden=""
  role="dialog"
  style="color:red"
  type="number"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#14-->
      <button
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > renders basic version 1`] = `
<div
  aria-modal="true"
  class="panel-dialog panel-dialog--mask-fade-slow"
  hidden=""
  role="dialog"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#14-->
      <button
        aria-label="close"
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      body content
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > renders basic version 2`] = `
<button
  aria-label="close"
  class="icon-btn panel-dialog__close"
  type="button"
>
  <svg
    aria-hidden="true"
    class="icon icon--16"
    focusable="false"
  >
    <defs>
      <symbol
        id="icon-close-16"
        viewBox="0 0 16 16"
      >
        <path
          d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
        />
      </symbol>
    </defs>
    <use
      href="#icon-close-16"
    />
  </svg>
</button>
`;

exports[`dialog > renders basic version 3`] = `
<div
  class="panel-dialog__main"
>
  <!--F#16-->
  body content
  <!--F/-->
</div>
`;

exports[`dialog > renders in open state 1`] = `
<div
  aria-modal="true"
  class="panel-dialog panel-dialog--mask-fade-slow"
  role="dialog"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#14-->
      <button
        aria-label="close"
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      body content
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > renders with end type 1`] = `
<div
  aria-modal="true"
  class="panel-dialog panel-dialog--mask-fade-slow"
  role="dialog"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide panel-dialog__window--end"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#14-->
      <button
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > renders with end type 2`] = `
<div
  class="panel-dialog__window panel-dialog__window--slide panel-dialog__window--end"
>
  <div
    class="panel-dialog__header"
  >
    <!--F#14-->
    <button
      class="icon-btn panel-dialog__close"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="icon icon--16"
        focusable="false"
      >
        <defs>
          <symbol
            id="icon-close-16"
            viewBox="0 0 16 16"
          >
            <path
              d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
            />
          </symbol>
        </defs>
        <use
          href="#icon-close-16"
        />
      </svg>
    </button>
    <!--F/-->
  </div>
  <div
    class="panel-dialog__main"
  >
    <!--F#16-->
    <!--F/-->
  </div>
</div>
`;

exports[`dialog > renders with header and footer 1`] = `
<div
  aria-labelledby="s0-0-dialog-title"
  aria-modal="true"
  class="panel-dialog panel-dialog--mask-fade-slow"
  hidden=""
  role="dialog"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#13-->
      <h2
        class="panel-dialog__title"
        id="s0-0-dialog-title"
      >
        <!--F#2-->
        title content
        <!--F/-->
      </h2>
      <!--F/-->
      <!--F#14-->
      <button
        aria-label="close"
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      body content
      <!--F/-->
    </div>
    <div
      class="panel-dialog__footer"
    >
      <!--F#20-->
      footer content
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > renders with header and footer 2`] = `
<button
  aria-label="close"
  class="icon-btn panel-dialog__close"
  type="button"
>
  <svg
    aria-hidden="true"
    class="icon icon--16"
    focusable="false"
  >
    <defs>
      <symbol
        id="icon-close-16"
        viewBox="0 0 16 16"
      >
        <path
          d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
        />
      </symbol>
    </defs>
    <use
      href="#icon-close-16"
    />
  </svg>
</button>
`;

exports[`dialog > renders with header and footer 3`] = `
<div
  class="panel-dialog__main"
>
  <!--F#16-->
  body content
  <!--F/-->
</div>
`;

exports[`dialog > renders with header and footer 4`] = `
<div
  class="panel-dialog__header"
>
  <!--F#13-->
  <h2
    class="panel-dialog__title"
    id="s0-0-dialog-title"
  >
    <!--F#2-->
    title content
    <!--F/-->
  </h2>
  <!--F/-->
  <!--F#14-->
  <button
    aria-label="close"
    class="icon-btn panel-dialog__close"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="icon icon--16"
      focusable="false"
    >
      <defs>
        <symbol
          id="icon-close-16"
          viewBox="0 0 16 16"
        >
          <path
            d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
          />
        </symbol>
      </defs>
      <use
        href="#icon-close-16"
      />
    </svg>
  </button>
  <!--F/-->
</div>
`;

exports[`dialog > renders with header and footer 5`] = `
<div
  class="panel-dialog__footer"
>
  <!--F#20-->
  footer content
  <!--F/-->
</div>
`;
